File: Reversal of words assignment/reversemultithreading.java
import java.util.Scanner;  // Import the Scanner class

public class reversemultithreading extends Thread {
    private String sentence;

    public reversemultithreading(String sentence) {
        this.sentence = sentence;
    }

    @Override
    public void run() {
        String[] words = sentence.split("\\s");
        StringBuilder sb = new StringBuilder();
        for (String word : words) {
            sb.append(new StringBuilder(word).reverse().toString() + " ");
        }
        sentence = sb.toString().trim();
    }

    public String getSentence() {
        return sentence;
    }

    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);
        System.out.print("Enter a string: ");  
        String str= sc.nextLine();   
        reversemultithreading thread = new reversemultithreading(str);
        thread.start();
        try {
            thread.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        String reversedSentence = thread.getSentence();
        System.out.println(reversedSentence);
    }
}

File: Reversal of words assignment/reversemultithreading.py
import threading

def reverseWord(word,lst):
    return lst.append(word[::-1])

para = input("Enter a sentence: ")

paraList = para.split(' ')
reversedParaList = []

threads = []
for word in paraList:
    thread = threading.Thread(target=reverseWord, args=(word,reversedParaList))
    threads.append(thread)

for thread in threads:
    thread.start()
    
for thread in threads:
    thread.join()

print(*reversedParaList)